planet/http.py

```diff
@@ -413,6 +414,34 @@ async def stream(
         finally:
             await response.aclose()
 
+    def client(self, name: str, base_url: Optional[str] = None) -> object:
+        """Get a client by its module name.
+
+        Parameters:
+            name: the name of the client module: data, orders, or
+                subscriptions.
+
+        Returns:
+            A client instance.
+
+        Raises:
+            ClientError when no such client can be had.
+
+        """
+        # To avoid circular dependency.
+        from planet.clients.data import DataClient
+        from planet.clients.orders import OrdersClient
+        from planet.clients.subscriptions import SubscriptionsClient
```

Comment:
Session and clients aren't fully decoupled because the client modules import `http`. We can fix that in the future. For now, we can tolerate some imports here. They don't have a significant impact on performance. By the time this method is called, those modules will already have been loaded.

Reply:
Do you want to file a ticket for fixing that in the future?

Reply:
No, not now. I'm not sure it's important. I changed the text above to reflect that.

Reply:
Note: the client modules import http in order to use http.Session in their type hints.

Reply:
I saw that!